bukkit/src/main/java/org/popcraft/bolt/listeners/BlockListener.java

```diff
@@ -413,17 +413,33 @@ public void onBlockIgnite(final BlockIgniteEvent e) {
 
     @EventHandler
     public void onPlayerBucketEmpty(final PlayerBucketEmptyEvent e) {
+        final Protection protection = plugin.findProtection(e.getBlock());
+        if (protection == null) {
+            return;
+        }
         final Player player = e.getPlayer();
-        if (!plugin.canAccess(e.getBlockClicked(), player, Permission.INTERACT) || !plugin.canAccess(e.getBlock(), player, Permission.DESTROY)) {
+        if (e.getBlock().equals(e.getBlockClicked()) && Tag.CAULDRONS.isTagged(e.getBlock().getType())) {
+            if (!plugin.canAccess(e.getBlock(), player, Permission.INTERACT, Permission.DEPOSIT)) {
```

Comment:
This could use the `canAccess(Protection, ...)` method to avoid finding the protection twice